plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -899,6 +899,9 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
         ImmutableMap.Builder<String, Optional<Partition>> resultBuilder = ImmutableMap.builder();
         for (Entry<String, List<String>> entry : partitionNameToPartitionValuesMap.entrySet()) {
             Partition partition = partitionValuesToPartitionMap.get(entry.getValue());
+            if (partition == null) {
```

Comment:
If the partition is gone during query planning, isn't that an error?

Reply:
Indeed. This is an error.
Would it make maybe more sense to log in the exception *ALL* the partitions which are missing ? Currently we just throw an exception for the first partition found missing.

Reply:
Where is the throwing code?

Reply:
https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java#L322-L325

Reply:
Why no throw here then?

BTW the code here seems to come from https://github.com/trinodb/trino/blob/1b73db6c67db8cd8b79fc7e68e4188f2c4facf46/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/BridgingHiveMetastore.java#L374, which seemingly was approved in  https://github.com/prestodb/presto/pull/4878, so cc @dain for more context

Reply:
The signature of the method

```
Map<String, Optional<Partition>> getPartitionsByNames(String databaseName, String tableName, List<String> partitionNames);
```

exists there from the first commit recorded on the `HiveMetastore`.